### ISSUE TITLE
Fix all broken references in docs by enabling nit-picky mode

### ISSUE THREAD

**Issue by kratsg** [scikit-hep/pyhf#1259]
This is needed and was missed in #1257 . The `-n` will be useful later when we need to check for broken links (atm, this is useless since we have too many unlinked `tensor` calls in the docs).

_Originally posted by @kratsg in https://github.com/scikit-hep/pyhf/pull/1258#discussion_r557722760_

```
-n
Run in nit-picky mode. Currently, this generates warnings for all missing references. See the config value nitpick_ignore for a way to exclude some references as “known missing”.
```

Reference: https://www.sphinx-doc.org/en/master/man/sphinx-build.html

**Comment by matthewfeickert** [comment]
@kratsg Once the `-n` is enabled then can fail on warnings for RTD also be turned on?

https://github.com/scikit-hep/pyhf/blob/75f3cd350ed3986d16d680fbb83f312791aafd68/.readthedocs.yml#L11

**Comment by kratsg** [comment]
> @kratsg Once the `-n` is enabled then can fail on warnings for RTD also be turned on?

Nope, because epub has a different set of warnings we'd need to fix.
